File: Teste com python/Questão3.py
import yaml

input = [
{ "name": "Maria Neusa de Aparecida", "cpf": "97905796671",
"state": "Sao Paulo", "value": "1234" },
{ "name": "Ricardo Fontes", "cpf": "44010762900",
"state": "Rio Grande do Sul", "value": "567" }
]
def solucao(arg):
# Retornar um string

    with open('# format-1.yaml') as f:
    
        docs = yaml.load_all(f, Loader=yaml.FullLoader)

        for doc in docs:
            for k, v in doc.items():
                c =arg[0:11]
                n =arg[11:25]
                val =arg[25::1]
                str1 = "cpf, name, value"
                str2 = (f"{c} | {n} | {int(val)}")
                keys = str1.split(", ") 
                values = str2.split("|") 
                dictionary = {} 
                for i in range(len(keys)): 
                    dictionary[keys[i]] = values[i] 
                print(dictionary) 
                break

solucao("97905796671Maria Neusa de00001234")

File: Mudanças/Questão 3.rb
require "yaml"

input = [
  { "name": "Maria Neusa de Aparecida", "cpf": "97905796671",
  "state": "Sao Paulo", "value": "1234" },
  { "name": "Ricardo Fontes", "cpf": "44010762900",
  "state": "Rio Grande do Sul", "value": "567" }
  ]

def solution(string)
  cpf =string[0,11]
  name =string[11,14]
  value =string[25,15] 
  format_string(cpf,name,value)
end 

def format_string(cpf,name,value)
  str1 = "cpf, name, value"
  str2 = ("#{cpf} | #{name} | #{value}")
  keys = str1.split(", ") 
  values = str2.split("|") 
  create_hash(keys, values, str1)
end

def create_hash(keys, values, str1)
  hash = {}
  while hash.size < 3
      for i in 0..2
        if keys[i] == "value"
          hash[keys[i]] = values[i].to_i
        else
          hash[keys[i]] = values[i] 
        end
  end
      end
      p hash
end
solution("44010762900Ricardo Fontes00000567")